rfcs/shared/build-system/triage-automation.md

```diff
@@ -0,0 +1,161 @@
+
+_Expected state:_
+
+- 1 label for v0
+- 1 label for v9
+- label per published package
+- "component labels" can stay as are but should be used solely for v8 (internal react package "sub-packages" https://github.com/microsoft/fluentui/labels/Component%3A%20Dialog)
+
+### 1. project label assignment
+
+> Note:
+>
+> - by "project" we mean one of libraries that exist within monorepo (v0,v8,v9,web-components)
+
+**Workflow:**
+
+Based on selected project within issue template, we should add appropriate label.
+
+<img width="1108" alt="image" src="https://user-images.githubusercontent.com/1223799/190106473-fd007d87-f09a-4493-aafd-781a8404ae45.png">
+
+> NOTE:
+>
+> - this is already implement via [github action](https://github.com/microsoft/fluentui/pull/24788)
+
+### 2. package label assignment
+
+> Note:
+>
+> - by "package" we mean official package name used to publish to npm registry without `@scope` prefix
+
+**Workflow:**
+
+Based on selected project within issue template, additional select should be dynamically generated so contributor can pick proper package.
+
+<img width="569" alt="example of flow" src="https://user-images.githubusercontent.com/1223799/190431945-56fcd343-6eb7-4145-b2b2-62f5919b8c31.png">
+
+Unfortunately **[dynamic input controls are not possible with github issues beta](https://github.com/community/community/discussions/29067)** which leaves us with following options how to implement this:
+
+- use only package names in "Library" picker
+  - cons
+    - huge list because we have quite a lot packages
+  - pros
+    - only 1 issue/feature template that we need to maintain
+    - removes need of `1. project label assignment` as labeling would be handled in one step
+- creating feature/bug templates per library
+  - cons
+    - 2 (feature/bug) x N templates (N is number of libraries in monorepo)
+  - pros
+    - clear separation that can be tweaked to particular library needs if needed
+- creating completely custom issue form solution (example [vuejs](https://github.com/vuejs/vue/issues/new/choose))
+  - cons
+    - big effort in terms of development and maintenance
+  - pros
+    - complete control about issue reporting
+    - another possible dog food for v9 controls
+
+**How will we maintain list of packages in issue template:**
+
+On library creation/removal we will tweak our issue/feature template automatically.
+
+> Note:
+>
+> - this can also happen only after created package has been published to npm registry ( post release GH action)
+
+### 3. assign issue to project board/team
+
+Based on assigned labels, we can automate assignment of issues based on our source of truth of ownership which is `CODEOWNERS` file.
+
+**Workflow:**
+
+```mermaid
+graph TD;
+    I("issue template created")-->L("labels added")-->C("parse codeowners");
+    subgraph "parse codeowners"
+    C-->D("find owner based on package")
+    end
+    D-->A("add issue to unified board")-->AA("pick team based on ownership definition");
```

Comment:
The role of the shield is to:
1. Verify the issue is actionable.
2. Set the priority for the issue.

In order to be able to do that, shield often needs to talk back to the reporter and/or debug the issue. As a side effect, the person on shield widen their knowledge to cover the whole library (useful for both support and development).

If the bot assigns the issues automatically, then:
1. We are siloing crews.
2. We require each crew to do their own "shield" (not necessarily having a dedicated person) to cover the two points above on a daily basis.

For that reason I would prefer the bot to suggest the owner but would still let shield to the check/assignment.

Reply:
> For that reason I would prefer the bot to suggest the owner but would still let shield to the check/assignment.

that's fine, it will still contain "needs triage" so person on shield can double check. we can create a view in project board that will show all "need triage" items for better visibility and post-bot processing. 

WDYT ?

Reply:
Works for me. I still think that shield is valuable and should visit every incoming issue for triage/investigation.
What you are proposing is achieving that while reducing manual repetitive work = win-win